app/src/main/java/org/mozilla/fenix/settings/WallpaperSettingsFragment.kt

```diff
@@ -53,54 +53,111 @@ class WallpaperSettingsFragment : Fragment() {
-                    for (wallpaper in wallpaperRow) {
-                        val thumbnailShape = RoundedCornerShape(8.dp)
-                        val border = if (currentlySelectedWallpaper == wallpaper) {
-                            Modifier.border(BorderStroke(2.dp, FirefoxTheme.colors.borderSelected), thumbnailShape)
-                        } else {
-                            Modifier
-                        }
-                        Box(
-                            modifier = Modifier
-                                .height(102.dp)
-                                .width(92.dp)
-                                .padding(4.dp)
-                                .clip(thumbnailShape)
-                                .then(border)
-                        ) {
-                            Image(
-                                painterResource(id = wallpaper.drawable),
-                                contentScale = ContentScale.FillBounds,
-                                contentDescription = wallpaper.name,
-                                modifier = Modifier
-                                    .fillMaxHeight()
-                                    .fillMaxWidth()
-                                    .clickable {
-                                        currentlySelectedWallpaper = wallpaper
-                                    }
-                            )
-                        }
-                    }
-                }
-            }
+    Column {
+        Surface {
+            WallpaperThumbnails(
+                wallpapers = wallpapers,
+                onSelectionChanged = { currentlySelectedWallpaper = it },
+                selectedWallpaper = currentlySelectedWallpaper
+            )
+        }
+        Row(
+            horizontalArrangement = Arrangement.SpaceBetween,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(horizontal = 16.dp, vertical = 24.dp)
+        ) {
+            var wallpapersEnabled by remember { mutableStateOf(false) }
+            Text("Logo wallpaper switch")
+            Switch(checked = wallpapersEnabled, onCheckedChange = { wallpapersEnabled = it })
+        }
+    }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun WallpaperThumbnails(
+    wallpapers: List<Wallpaper>,
+    onSelectionChanged: (Wallpaper) -> Unit,
+    selectedWallpaper: Wallpaper
+) {
+    LazyVerticalGrid(
+        cells = GridCells.Fixed(3),
+        modifier = Modifier.padding(vertical = 30.dp, horizontal = 20.dp)
+    ) {
+        items(wallpapers) { wallpaper ->
+            WallpaperThumbnailItem(
+                wallpaper = wallpaper,
+                onSelectionChanged = onSelectionChanged,
+                isCurrentlySelected = selectedWallpaper == wallpaper
+            )
+        }
+    }
+}
+
+@Composable
+fun WallpaperThumbnailItem(
+    wallpaper: Wallpaper,
+    onSelectionChanged: (Wallpaper) -> Unit,
+    isCurrentlySelected: Boolean
```

Comment:
Maybe to be less verbose, I would just rename this to `isSelected`.

Reply:
Yeah good call. Updated